base_layer/wallet/tests/output_manager_service/service.rs

```diff
@@ -337,6 +337,231 @@ fn fee_estimate_memory_db() {
     fee_estimate(OutputManagerMemoryDatabase::new());
 }
 
+pub fn setup_oms_with_bn_state<T: OutputManagerBackend + 'static>(
+    runtime: &mut Runtime,
+    backend: T,
+    height: Option<u64>,
+) -> (
+    OutputManagerHandle,
+    Shutdown,
+    TransactionServiceHandle,
+    BaseNodeServiceHandle,
+)
+{
+    let shutdown = Shutdown::new();
+    let factories = CryptoFactories::default();
+
+    let (oms_request_sender, oms_request_receiver) = reply_channel::unbounded();
+    let (oms_event_publisher, _) = broadcast::channel(200);
+
+    let (ts_request_sender, _ts_request_receiver) = reply_channel::unbounded();
+    let (event_publisher, _) = channel(100);
+    let ts_handle = TransactionServiceHandle::new(ts_request_sender, event_publisher.clone());
+
+    let constants = ConsensusConstantsBuilder::new(Network::Stibbons).build();
+
+    let (sender, receiver_bns) = reply_channel::unbounded();
+    let (event_publisher_bns, _) = broadcast::channel(100);
+
+    let base_node_service_handle = BaseNodeServiceHandle::new(sender, event_publisher_bns);
+    let mut mock_base_node_service = MockBaseNodeService::new(receiver_bns, shutdown.to_signal());
```

Comment:
I like the update to the Mock and this is more a comment for the future cause the tests in this PR seem fine but when making a Mock like this I like to separate the state into its own thread-safe struct so that we can manipulate it during the test. For example check out the RPC mock https://github.com/tari-project/tari/blob/development/base_layer/wallet/tests/support/rpc.rs#L77.

This allows you to do things like have the BN service be unconnected and then change to connected later on.

Reply:
Thanks for the feedback, that would definitely be cleaner. Still getting the hang of this testing thing lol.